django-stubs/db/models/fields/__init__.pyi

```diff
@@ -181,16 +181,16 @@ class Field(RegisterLookupMixin, Generic[_ST, _GT]):
         validators: Iterable[validators._ValidatorCallable] = ...,
         error_messages: Optional[_ErrorMessagesT] = ...,
     ): ...
-    def __set__(self, instance, value: _ST) -> None: ...
+    def __set__(self, instance: Any, value: _ST) -> None: ...
     # class access
     @overload
-    def __get__(self: _T, instance: None, owner) -> _T: ...
+    def __get__(self: _T, instance: None, owner: Any) -> _T: ...
     # Model instance access
     @overload
-    def __get__(self, instance: Model, owner) -> _GT: ...
+    def __get__(self, instance: Model, owner: Any) -> _GT: ...
     # non-Model instances
     @overload
-    def __get__(self: _T, instance, owner) -> _T: ...
+    def __get__(self: _T, instance: Any, owner: Any) -> _T: ...
```

Comment:
I'm unsure what these `owner` fields are.

Reply:
I think it's `type[M] | None = None` (if not in pyi)

Where

```
M = TypeVar("M", bound=Model)
```

And I think it's not none when instance is not none.

Making this an `@overload` function

Reply:
So in this specific overload both any's should be `: None` I think

Reply:
@flaeppe You're probably right but I'm wary of touching these since I don't totally understand what is happening here. Are you up for opening a follow-up PR?

Reply:
I think it should be fine keeping it as `Any` since what's exposed as an API is `_T`. The `instance` and `owner` types would only be relevant for the implementation body, but this is in stubs so we won't be affected here.

The only time they'd come in to play and have an impact would be while/if subclassing `Field` and overriding it's descriptor methods. If we keep `Any`, we'll allow a child class to declare them as whatever type they deem fit, while doing e.g. `type[M] | None` we'll restrain it.

Though, in general I don't see any obvious use cases for overriding the descriptor for `Field`, but I might be wrong here. The normal Django use case would be to implement a separate field descriptor class (https://docs.djangoproject.com/en/4.1/ref/models/fields/#django.db.models.Field.descriptor_class) to decide what goes in and out of a model field